grommet/hpe-design-system
# 2733 Design Site In-Page Table of Contents

[Deploy Preview](https://deploy-preview-2753--keen-mayer-a86c8b.netlify.app/components/card)

#### What does this PR do?
Adds a table of contents to component pages that allows the user to quickly navigate to all sections headers on the page. The ToC is "sticky" and will not disappear from view when the user scrolls through the page. It also highlights the current section the user is on for added visibility. Uses regex to capture and parse all markdown pages. On certain pages with no headings (ex: Box), the table of contents is not shown. Final design to indicate active section header in the ToC TBD.

#### Where should the reviewer start?
aries-site/src/layouts/content/InPageNavigation.js
aries-site/src/layouts/main/Layout.js
aries-site/src/scripts/site-contents.mjs

#### What testing has been done on this PR?
Manual testing on all component pages to ensure all headings are correctly found by regex and displayed on table of contents.

In addition to the feature you are implementing, have you checked the following:

**General UX Checks**

**Accessibility Checks**

**Code Quality Checks**

#### How should this be manually tested?
Click heading links on table of contents, ensure that all headings show up on each component page.

#### Any background context you want to provide?
Existing table of contents branch for reference at #2134, this branch builds off of it and implements further details that conform to DS template guide. Currently many component pages have a lot of information that is not easily accessible and requires a lot of scrolling in a single page, and the table of contents is the first step towards making standardized and accessible documentation. There was a formatting issue with some of the markdown pages (with CardGrid component) that caused the content to be crushed into each other, but this had to do with how src/components/cards/CardGrid.js was filling the container and has been fixed.

#### What are the relevant issues?
Closes #2734.

#### Screenshots (if appropriate)
Most updated designs:

<img width="1381" alt="image" src="https://user-images.githubusercontent.com/23144209/183139152-d3ce1ebe-6866-43c4-88cb-924ae61c3e03.png">

<img width="1390" alt="image" src="https://user-images.githubusercontent.com/23144209/183139233-04db609a-ab54-4223-b13b-cfddb9b818e3.png">

<img width="1367" alt="image" src="https://user-images.githubusercontent.com/23144209/183139303-94483be7-2731-4651-adb7-c69531a29a96.png">

#### Should this PR be mentioned in Design System updates?
Yes - Add table of contents to the Design System content pages.

#### Is this change backwards compatible or is it a breaking change?
Backwards compatible.
